e2e/e2etests/test_spl_withdraw.go

```diff
@@ -0,0 +1,69 @@
+package e2etests
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/gagliardetto/solana-go"
+	"github.com/gagliardetto/solana-go/rpc"
+	"github.com/stretchr/testify/require"
+
+	"github.com/zeta-chain/node/e2e/runner"
+)
+
+func TestSPLWithdraw(r *runner.E2ERunner, args []string) {
+	require.Len(r, args, 1)
+
+	withdrawAmount := parseBigInt(r, args[0])
+
+	// get SPL ZRC20 balance before withdraw
+	zrc20BalanceBefore, err := r.SPLZRC20.BalanceOf(&bind.CallOpts{}, r.EVMAddress())
+	require.NoError(r, err)
+	r.Logger.Info("runner balance of SPL before withdraw: %d", zrc20BalanceBefore)
+
+	require.Equal(r, 1, zrc20BalanceBefore.Cmp(withdrawAmount), "Insufficient balance for withdrawal")
+
+	// parse withdraw amount (in lamports), approve amount is 1 SOL
+	approvedAmount := new(big.Int).SetUint64(solana.LAMPORTS_PER_SOL)
+	require.Equal(
+		r,
+		-1,
+		withdrawAmount.Cmp(approvedAmount),
+		"Withdrawal amount must be less than the approved amount (1e9)",
+	)
+
+	// load deployer private key
+	privkey, err := solana.PrivateKeyFromBase58(r.Account.SolanaPrivateKey.String())
+	require.NoError(r, err)
+
+	// get receiver ata balance before withdraw
+	receiverAta := r.FindOrCreateAssociatedTokenAccount(privkey, privkey.PublicKey(), r.SPLAddr)
```

Comment:
nit: Not really an issue but since this function is related only to Solana, could it be renamed somehow, like `SolanaFindOrCreateATA`. In the end something that shows it's Solana related, to group functions by chain and makes it easier in IDEs to filter for chain functions.